File: ch2/exponent_by_iteration.rb
def exponent_by_iteration(a, n)
  result = 1
  # for i in (1..n)
  1.upto n do
    result *= a
  end
  result
end

p exponent_by_iteration(3, 6) # => 729
p exponent_by_iteration(10, 3) # => 1000
p exponent_by_iteration(17, 10) # => 2015993900449

File: ch2/exponentByIteration.py
def exponentByIteration(a, n):
  result = 1
  for i in range(n):
    result *= a
  return result

print(exponentByIteration(3, 6)) # => 729
print(exponentByIteration(10, 3)) # => 1000
print(exponentByIteration(17, 10)) # => 2015993900449